### PR TITLE
[9.0][NEW] stock_deposit: Manage deposit locations

### PR DESCRIPTION
**Pull request by sergio-teruel**
New base module to allow manage deposit locations. You can transfer quatities from one location to other deposit location and return products from deposit location to stock location.
Also you can regularize quants, the wizard makes a picking from deposit location to customer locations.

cc @Tecnativa

### CODE REVIEW

File: stock_deposit/models/product.py
Comment (Cedric-Pigeon) [inline stock_deposit/models/product.py]: Please avoid to rename safe_eval by eval, it can be disturbing

File: stock_deposit/models/product.py
Comment (Cedric-Pigeon) [inline stock_deposit/models/product.py]: Do not use eval. Use res_action['context'].copy() instead

File: stock_deposit/models/product.py
Comment (Cedric-Pigeon) [inline stock_deposit/models/product.py:45]: This way of computing stock field can be source of performance problem.
You should:
1/ extend _product_available and complete the returned dictionary with your value
2/ call it once in _compute_deposit_available and retrieve the result in the dictionary

File: stock_deposit/models/stock.py
Comment (Cedric-Pigeon) [inline stock_deposit/models/stock.py]: do not use camel case for variable name

File: stock_deposit/README.rst
Comment (pedrobaeza) [inline stock_deposit/README.rst]: s/type/types

Comment (pedrobaeza) [reply stock_deposit/README.rst]: s/out deposits/outgoing deposits

Comment (pedrobaeza) [reply stock_deposit/README.rst]: s/return deposits/returned deposits

File: stock_deposit/README.rst
Comment (pedrobaeza) [inline stock_deposit/README.rst]: Put the `*` after `Owners` word

File: stock_deposit/README.rst
Comment (pedrobaeza) [inline stock_deposit/README.rst]: s/of product type/of type "Stockable"

File: stock_deposit/README.rst
Comment (pedrobaeza) [inline stock_deposit/README.rst]: s/deposit out/"Outgoing deposits"

File: stock_deposit/README.rst
Comment (pedrobaeza) [inline stock_deposit/README.rst]: comma after `process`

Comment (pedrobaeza) [reply stock_deposit/README.rst]: Remove `de` word

Comment (pedrobaeza) [reply stock_deposit/README.rst]: Put this also as another point with `#.`

File: stock_deposit/README.rst
Comment (pedrobaeza) [inline stock_deposit/README.rst]: Start a new point with `#.` for this and put `You can also see the deposits for all your products in the menu entry Inventory --> Reports --> Deposit location inventory`

File: stock_deposit/README.rst
Comment (pedrobaeza) [reply stock_deposit/README.rst]: And start the "section" with something like: `To track current deposits:"

File: stock_deposit/README.rst
Comment (pedrobaeza) [inline stock_deposit/README.rst]: s/regularize/regularizing

File: stock_deposit/README.rst
Comment (pedrobaeza) [inline stock_deposit/README.rst]: s/Go to/To go

Comment (pedrobaeza) [reply stock_deposit/README.rst]: s/you want/you want to

File: stock_deposit/README.rst
Comment (pedrobaeza) [reply stock_deposit/README.rst]: s/deposit/deposited

File: stock_deposit/README.rst
Comment (pedrobaeza) [inline stock_deposit/README.rst]: s/deposit/deposited

Comment (pedrobaeza) [reply stock_deposit/README.rst]: Point instead of comma.

Comment (pedrobaeza) [reply stock_deposit/README.rst]: s/create/creates

File: stock_deposit/README.rst
Comment (pedrobaeza) [inline stock_deposit/README.rst]: s/out pickings/delivery orders

File: stock_deposit/README.rst
Comment (pedrobaeza) [reply stock_deposit/README.rst]: s/by each warehouse/per warehouse

File: stock_deposit/data/stock_deposit_data.xml
Comment (pedrobaeza) [inline stock_deposit/data/stock_deposit_data.xml:7]: Why this sequence? It should use the delivery order picking type sequence, or you will find under the same card pickings with different sequences. Or do you prefer this way?

File: stock_deposit/README.rst
Comment (pedrobaeza) [reply stock_deposit/README.rst]: Define first what is a "regularization", that means to deliver deposited stock to the customer as a normal outgoing move.

File: stock_deposit/README.rst
Comment (pedrobaeza) [inline stock_deposit/README.rst]: `.` at the end

Comment (pedrobaeza) [reply stock_deposit/README.rst]: You should mention that deposited stocks can't be selected in regular delivery orders unless they have been returned.

Also tell about the inventory valuation.

File: stock_deposit/models/stock.py
Comment (pedrobaeza) [inline stock_deposit/models/stock.py]: Don't make an if statement:

```
picking_type.is_deposit = (
    picking_type.default_location_dest_id.deposit_location or
    picking_type.default_location_src_id.deposit_location
)
```

File: stock_deposit/models/stock.py
Comment (pedrobaeza) [inline stock_deposit/models/stock.py]: Please, one model per file, as stated in https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md

File: stock_deposit/models/stock.py
Comment (pedrobaeza) [inline stock_deposit/models/stock.py]: Put `self.ensure_one()`

File: stock_deposit/models/stock.py
Comment (pedrobaeza) [inline stock_deposit/models/stock.py]: Make this translatable with `_()`

File: stock_deposit/models/stock.py
Comment (pedrobaeza) [inline stock_deposit/models/stock.py]: Put `self.ensure_one()`

File: stock_deposit/models/stock.py
Comment (pedrobaeza) [inline stock_deposit/models/stock.py]: Put `self.ensure_one()`

File: stock_deposit/views/stock_dashboard.xml
Comment (pedrobaeza) [inline stock_deposit/views/stock_dashboard.xml]: Don't put name. It's automatically derived.

File: stock_deposit/views/stock_view.xml
Comment (pedrobaeza) [inline stock_deposit/views/stock_view.xml]: Deposited Quants

File: stock_deposit/views/stock_view.xml
Comment (pedrobaeza) [inline stock_deposit/views/stock_view.xml]: Deposited Inventory Valuation

File: stock_deposit/views/stock_view.xml
Comment (pedrobaeza) [inline stock_deposit/views/stock_view.xml]: Don't make this format, as this makes the view unreadable in Odoo. Put it in one line.

File: stock_deposit/views/stock_view.xml
Comment (pedrobaeza) [inline stock_deposit/views/stock_view.xml]: Why not call the XML-ID as standard: `action_deposit_quant`?

File: stock_deposit/wizards/stock_quant_wizard.py
Comment (pedrobaeza) [inline stock_deposit/wizards/stock_quant_wizard.py]: s/not/non (update also translation)

File: stock_deposit/wizards/stock_quant_wizard.py
Comment (pedrobaeza) [inline stock_deposit/wizards/stock_quant_wizard.py]: If you put a prepare method, use one per quant

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

**Comment by pedrobaeza** [comment]
For the icon, you can use this icon: https://openclipart.org/detail/168751/saving-up, replacing the coin with the old warehouse icon: https://github.com/odoo/odoo/blob/8.0/addons/stock/static/description/icon.png

**Comment by pedrobaeza** [comment]
Another thing: the deposit picking types colors should be different from the warehouse ones (one that you decide, but not the same).

**Review by rafaelbn — COMMENTED** [review]
I've tested in runbot.
- I update iPad Mini qty to 5000
- I made a deposit of 500 but when I click in "reserve" it doesn't take quantity it never change to state "available"

Please review, thanks.

File: stock_deposit/data/stock_deposit_data.xml
Comment (sergio-teruel) [reply stock_deposit/data/stock_deposit_data.xml:7]: I thought this. We can a sequence for regularized pickings

File: stock_deposit/views/stock_dashboard.xml
Comment (sergio-teruel) [reply stock_deposit/views/stock_dashboard.xml]: For all inherited views??
But in this way i localize this view more easily, i think..

Comment (pedrobaeza) [reply stock_deposit/views/stock_dashboard.xml]: Well, for me it's not helpful, but it's your choice. At least correct typo :wink:

**Comment by sergio-teruel** [comment]
All picking types color are the same that your warehouse... so I think it's better to have this color...

**Review by Cedric-Pigeon — CHANGES_REQUESTED** [review]

**Comment by sergio-teruel** [comment]
@rafaelbn REadme file says that you must active the owner option in inventory setting and a normal action is go to kanban card and click in 'Deposit Out' and make a deposit picking, the origin location must be have enought product qty's.

**Comment by sergio-teruel** [comment]
@Cedric-Pigeon Changes done

File: stock_deposit/models/product.py
Comment (pedrobaeza) [reply stock_deposit/models/product.py:45]: The dictionary cannot be completed as there's no prepare method for this dictionary, but the implementation in https://github.com/OCA/stock-logistics-workflow/pull/268/commits/96d98054debbf09337d93c834fdcd222432b91fd is more efficient than previous one.

**Review by pedrobaeza — APPROVED** [review]
(code review) A functional is desired cc @rafaelbn

**Review by Cedric-Pigeon — APPROVED** [review]
Code review

File: stock_deposit/README.rst
Comment (rafaelbn) [inline stock_deposit/README.rst]: s/"Outgoing deposits"/"deposit out"

File: stock_deposit/README.rst
Comment (rafaelbn) [inline stock_deposit/README.rst]: s/Deposit location inventory/Deposited Inventory Valuation

File: stock_deposit/README.rst
Comment (rafaelbn) [inline stock_deposit/README.rst]: If here is s/deposit inventory/Deposited Inventory Valuation , where is the report? I not, I cannot find this menú item.

File: stock_deposit/README.rst
Comment (rafaelbn) [inline stock_deposit/README.rst:53]: Please explain now how to give back deposit to our warehouse:

> To give back deposit to you warehouse just make standard process, go to deposit and click in buton "Reverse"

And now here there is a BUG because when you reverse a deposit it doesn't go to "Deposit in"

File: stock_deposit/__openerp__.py
Comment (rafaelbn) [inline stock_deposit/__openerp__.py]: Only "Tecnativa" without S.L.

**Review by rafaelbn — CHANGES_REQUESTED** [review]
Thanks for this exceptional module!

**Comment by sergio-teruel** [comment]
@rafaelbn Changes done. Deposit in is only a picking type, so you can do a picking from deposit location to stock.

**Review by rafaelbn — APPROVED** [review]
Again tested

**Comment by pedrobaeza** [comment]
@sergio-teruel, one last thing before merging about the icon: please include SVG for further editing, include a link to the original page in the README, and please make bigger (a lot bigger) the box icon, because it cannot be seen right now and it's an important element in the icon.

**Comment by sergio-teruel** [comment]
@pedrobaeza Changes done.

**Pull request merged**
